lib/simpler/router/route.rb

```diff
@@ -12,9 +12,30 @@ def initialize(method, path, controller, action)
       end
 
       def match?(method, path)
-        @method == method && path.match(@path)
+        @method == method && path_match?(path)
       end
 
+      def variables(path)
+        path_request = split_path(path)
+        path_route = split_path(@path)
+        variables = {}
+        path_route.zip(path_request).each { |route, request| variables[route[1..-1].to_sym] = request if route[0] == ':' }
+        variables
+      end
+
+      private
+
+      def path_match?(path)
+        path_request = split_path(path)
+        path_route = split_path(@path)
+        return false if path_request.count != path_route.count
+        path_route.zip(path_request).each { |route, request| return false if route[0] != ':' && request != route }
+        true
+      end
+
+      def split_path(path)
+        path.split('/') - ['']
```

Comment:
Так понятнее что происходит
```suggestion
        path.split('/')[1..-1]
```

Reply:
Я так делал, чтобы пути //tests//1 и /tests/:id считались одинаковыми. Проверял в браузере на разных сайтах - подряд идущие слэши не приводят к ошибке и обрабатываются как один слэш.

Reply:
Ну тогда твой метод не работает, потому что он не все пустые части выбросит. Надо `path.split('/').reject(&:empty?)`

Reply:
А почему не все? Я пробовал – он выкидывал вроде бы все. (сейчас еще раз проверил – все выкидывает)